app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -264,35 +261,31 @@ async function addEthereumChainHandler(
       }),
     );
   }
-
+  let networkConfigurationId;
   try {
-    await addCustomRpc(
-      await requestUserApproval({
-        origin,
-        type: MESSAGE_TYPE.ADD_ETHEREUM_CHAIN,
-        requestData: {
-          chainId: _chainId,
-          blockExplorerUrl: firstValidBlockExplorerUrl,
-          chainName: _chainName,
-          rpcUrl: firstValidRPCUrl,
-          ticker,
-        },
-      }),
-    );
-
-    sendMetrics({
```

Comment:
Looks like this metric event has been removed. Is that happening somewhere else now?

Reply:
Its in the `upsertNetworkConfiguration` call itself: https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6R4383

Reply:
Hmm, I see. I saw that we were still calling `upsertNetworkConfiguration` directly from the UI as well, so I was concerned this would result in this event being tracked twice. But it looks like it's only dispatched in the UI for internal add network requests, so we should be good.

Reply:
Wait no, it's still being duplicated. This event is tracked directly in the UI as well.

Reply:
Where exactly are you seeing the duplication?

Reply:
The `upsertNetworkConfiguration` function is called here, and it's called in the UI in the network form. These are the two places a custom network can be added. This event was dispatched in each place separately. Search for `Custom Network Added` to find the other event in the UI still being dispatched.

Reply:
Yeah. Seems simplest to leave it here

Reply:
So I actually refactored to move the `trackEvent` into the `NetworkController` as we'd discussed previously. So now if its not editing an existing configuration it will call this event: https://github.com/MetaMask/metamask-extension/pull/17421/commits/bbf9c71fc6e6403d98e18a3efec5933796e74f98#diff-b3422fb8c3cd155660c8689ba9fdd32ca0dcfe799c152a10be2b8c05d47310ceR577-R578